src/rdkafka_partition.c

```diff
@@ -128,7 +128,7 @@ static void rd_kafka_toppar_consumer_lag_req (rd_kafka_toppar_t *rktp) {
         if (rktp->rktp_wait_consumer_lag_resp)
                 return; /* Previous request not finished yet */
 
-        rkb = rd_kafka_toppar_leader(rktp, 1/*proper brokers only*/);
+        rkb = rd_kafka_toppar_broker(rktp, 1/*proper brokers only*/);
```

Comment:
Do we want to base the consumer lag on the replica's view of the High-Watermark or the leader's?

Reply:
this is a good question. an even better question is will it even work, given the Offsets API - at least before FFF was implemented - requires the request to be directed at the leader. it turns out this is still the case, and this doesn't even work.

can you think of anything else that uses rktp_broker that may also break? I will check as well.